src/extension/src/azure/azure-resource-group/resourceGroupModule.ts

```diff
@@ -0,0 +1,96 @@
+import { ResourceManagementClient } from "azure-arm-resource/lib/resource/resourceManagementClient";
+import { ResourceGroupError, AuthorizationError } from "../../errors";
+import { CONSTANTS } from "../../constants";
+
+const VALIDATION_LIMIT = 3;
+
+export interface ResourceGroupSelection {
+    subscriptionItem: SubscriptionItem;
+    resourceGroupName: string;
+    location: string;
+}
+
+export class ResourceGroupDeploy {
+    private azureResourceClient: ResourceManagementClient | undefined;
+
+    private setAzureResourceClient (userSubscriptionItem: SubscriptionItem) {
+        if (
+            this.azureResourceClient === undefined ||
+            this.azureResourceClient.subscriptionId !== userSubscriptionItem.subscriptionId
+            ) {
+                this.azureResourceClient = new ResourceManager().getResourceManagementClient(userSubscriptionItem);
+            }
+    }
+
+    public async createResourceGroup(
+        resourceGroupSelection: ResourceGroupSelection
+    ): Promise<string | undefined> {
+        const {subscriptionItem, resourceGroupName, location} = resourceGroupSelection;
+
+        let azureResourceClient: ResourceManagementClient = new ResourceManager().getResourceManagementClient(subscriptionItem);
+
+        let options: ResourceGroup = {
+            name: resourceGroupName,
+            location: location,
+        };
+
+        let result = await azureResourceClient.resourceGroups.createOrUpdate(
+            resourceGroupName, 
+            options
+            ); 
+        return result.name;
+    }
+
+    public async generateValidResourceGroupName(
+        name: string, 
+        userSubscriptionItem: SubscriptionItem
+    ): Promise<string> {
+        let tries = 0;
+        this.setAzureResourceClient(userSubscriptionItem);
+        let generatedName = this.generateResourceGroupName(name);
+        let isValid = await this.validateResourceGroupName(generatedName);
+
+        while (tries < VALIDATION_LIMIT && !isValid) {
+            generatedName = this.generateResourceGroupName(name);
+            isValid = await this.validateResourceGroupName(generatedName);
+            tries++;
+        }
+        if (tries >= VALIDATION_LIMIT) {
+            throw new ResourceGroupError(CONSTANTS.ERRORS.RESOURCE_GROUP_TRIES_EXCEEDED);
+        }
+        return generatedName;
+    }
+
+    private async validateResourceGroupName(
+        name: string
+    ): Promise<boolean> {
+        if (this.azureResourceClient === undefined) {
+            throw new AuthorizationError(CONSTANTS.ERRORS.AZURE_RESOURCE_CLIENT_NOT_DEFINED);
+        }
+        let exist = await this.azureResourceClient.resourceGroups.checkExistence(name);
+        return await !exist;
+    }
+
+    private generateResourceGroupName(userProjectName: string): string {
+        let unixTimestamp = Date.now();
+        let suffix = this.unixToSuffix(unixTimestamp);
+        return userProjectName + "_" + suffix;
```

Comment:
Apparently the max length of a resource group is 90 characters. 
![image](https://user-images.githubusercontent.com/32103099/60742569-e081ca80-9f3b-11e9-9ce3-39c870e5966e.png)

What happens if the project name is > 81 characters? 

Look at this snippit of the library code for more info:

https://github.com/Azure/azure-sdk-for-node/blob/40bad8fd52c0cc2264a0900a4d412c8f68aef263/lib/services/resourceManagement/lib/resource/operations/resourceGroups.js#L40-L80

Reply:
Thanks @SahilTara, totally did not think about this. I'll find a way to handle this after talking to the team.

Reply:
My suggestion would be having a max project length of 75 if you want to keep that suffix.

Reply:
Exactly our thoughts, the change is now in PR (https://github.com/microsoft/WebTemplateStudio/pull/836)